regional_mom6/regional_mom6.py

```diff
@@ -1099,10 +1561,134 @@ def simple_boundary(
             repeat_year_forcing=self.repeat_year_forcing,
         )
 
-        seg.rectangular_brushcut()
+        seg.regrid_rectangle_tracers()
+
+        # Save Segment to Experiment
+        self.segments[orientation] = seg
         print("Done.")
         return
 
+    def setup_tides_boundaries(
+        self,
+        path_to_td,
+        tidal_filename,
+        tidal_constituents="read_from_expt_init",
+        boundary_type="rectangle",
+    ):
+        """
+        This function:
+        We subset our tidal data and generate more boundary files!
+
+        Args:
+            path_to_td (str): Path to boundary tidal file.
+            tidal_filename: Name of the tpxo product that's used in the tidal_filename. Should be h_{tidal_filename}, u_{tidal_filename}
+            tidal_constiuents: List of tidal constituents to include in the regridding. Default is [0] which is the M2 constituent.
+            boundary_type (Optional[str]): Type of boundary. Currently, only ``'rectangle'`` is supported. Here 'rectangle' refers to boundaries that are parallel to lines of constant longitude or latitude.
+        Returns:
+            *.nc files: Regridded tidal velocity and elevation files in 'inputdir/forcing'
+
+        General Description:
+        This tidal data functions are sourced from the GFDL NWA25 and changed in the following ways:
+         - Converted code for RM6 segment class
+         - Implemented Horizontal Subsetting
+         - Combined all functions of NWA25 into a four function process (in the style of rm6) (expt.setup_tides_rectangular_boundaries, segment.coords, segment.regrid_tides, segment.encode_tidal_files_and_output)
+
+
+        Original Code was sourced from:
+        Author(s): GFDL, James Simkins, Rob Cermak, etc..
+        Year: 2022
+        Title: "NWA25: Northwest Atlantic 1/25th Degree MOM6 Simulation"
+        Version: N/A
+        Type: Python Functions, Source Code
+        Web Address: https://github.com/jsimkins2/nwa25
+        """
+        if boundary_type != "rectangle":
+            raise ValueError(
+                "Only rectangular boundaries are supported by this method."
+            )
+        if tidal_constituents != "read_from_expt_init":
+            self.tidal_constituents = tidal_constituents
+        tpxo_h = (
+            xr.open_dataset(os.path.join(path_to_td, f"h_{tidal_filename}"))
+            .rename({"lon_z": "lon", "lat_z": "lat", "nc": "constituent"})
+            .isel(
+                constituent=convert_to_tpxo_tidal_constituents(self.tidal_constituents)
+            )
+        )
+        tidal_360_lon = [
+            self.longitude_extent[0],
+            self.longitude_extent[1],
+        ]
+        ny0, nx0 = find_roughly_nearest_ny_nx(
```

Comment:
This line crashed when testing with the NWA run. Perhaps an issue with the curved boundary? I'm looking into it

Reply:
Deleted b/c not necessarily needed, added to next PR